node/core/dispute-coordinator/src/scraping/mod.rs

```diff
@@ -200,7 +207,23 @@ impl ChainScraper {
 		{
 			Some(key_to_prune) => {
 				self.backed_candidates.remove_up_to_height(&key_to_prune);
-				self.included_candidates.remove_up_to_height(&key_to_prune);
+				let candidates_pruned = self.included_candidates.remove_up_to_height(&key_to_prune);
+				// Removing entries from inclusions per candidate referring to blocks before 
+				// key_to_prune
+				for candidate_hash in candidates_pruned {
+					self.inclusions_per_candidate.entry(candidate_hash).and_modify(|inclusions| {
+						*inclusions = inclusions
+							.clone()
+							.into_iter()
+							.filter(|inclusion| inclusion.0 >= key_to_prune)
+							.collect::<HashSet<(BlockNumber, Hash)>>();
```

Comment:
nit: I believe this can be simplified by using [`retain`](https://doc.rust-lang.org/1.65.0/std/collections/struct.HashSet.html#method.retain). 🙂